### PR TITLE
GH-1362: Enable Capture of Tx Synchronization Fail

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-amqp/issues/1362

**cherry-pick to 2.3.x**

### CODE REVIEW

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java [hunk 1]
```diff
@@ -638,6 +642,9 @@ void resourcesClearedAfterTxFailsWithSync() throws IOException, TimeoutException
 		assertThatIllegalStateException()
 			.isThrownBy(() -> (TransactionSynchronizationManager.getSynchronizations()).isEmpty())
 			.withMessage("Transaction synchronization is not active");
+		assertThatExceptionOfType(AfterCompletionFailedException.class)
+			.isThrownBy(() -> ConnectionFactoryUtils.checkAfterCompletion());
+		ConnectionFactoryUtils.enableAfterCompletionFailureCapture(true);
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java:647]: You probably mean `false` in the end of test.
Plus I believe you realize that this is global static flag, which can be changed in one thread, but checked in the other.
Is it really intentional?
Also: you probably have missed to mention this flag in the doc.

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java (hunk 1 above)
Comment (garyrussell) [reply spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java:647]: Oops; yes; it was intentionally global - they shouldn't have to set it on each thread.

Doh - I had an unsaved change to the doc file.

Comment (artembilan) [reply spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java:647]: Good, but why this is still `true` in the end of this test ? 😄

**Review by artembilan — CHANGES_REQUESTED** [review]

**Comment by artembilan** [comment]
... and cherry-picked to `2.3.x`

**Pull request merged**